components/director/internal/formationmapping/auth_middleware.go

```diff
@@ -172,8 +190,92 @@ func (a *Authenticator) Handler() func(next http.Handler) http.Handler {
 	}
 }
 
-// isAuthorized verify through custom logic the caller is authorized to update the formation assignment status
-func (a *Authenticator) isAuthorized(ctx context.Context, formationAssignmentID, formationID string) (bool, int, error) {
+// FormationHandler is a handler middleware that executes authorization check for the formation requests reporting status
+func (a *Authenticator) FormationHandler() func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+
+			if r.Method != http.MethodPatch {
+				w.WriteHeader(http.StatusMethodNotAllowed)
+				return
+			}
+
+			routeVars := mux.Vars(r)
+			formationID := routeVars[FormationIDParam]
+
+			if formationID == "" {
+				log.C(ctx).Errorf("Missing required parameters: %q", FormationIDParam)
+				respondWithError(ctx, w, http.StatusBadRequest, errors.New("Not all of the required parameters are provided"))
```

Comment:
I haven't included it because the parameter in the code(`ucl-formation-id`) is slightly different from the one in our swagger documentation(`uclFormationId `). And if we include it in the error message then we need to take care when the swagger is updated to update our code too. We could forget, and the error message could bring more confusion than help.